### ISSUE TITLE
Guard against billing surprises - DatabaseMigrations sample

### ISSUE THREAD

**Issue by collinbarrett** [dotnet/aspire-samples#300]
The `DatabaseMigrations` sample is working great for me in local development. However, by default, it's included when I `azd up`. The `MigrationService` seems to get deployed as an Azure Container App 'App' type rather than 'Job' type ([docs](https://learn.microsoft.com/en-us/azure/container-apps/jobs?tabs=azure-cli#compare-container-apps-and-jobs)), so Container Apps continually reboots it even if the migration completes successfully.

I discovered this when I looked at my Azure SQL Database [free offer](https://learn.microsoft.com/en-us/azure/azure-sql/database/free-offer?view=azuresql) quota. Stopping the `MigrationService` container in Azure Portal caused the inflection in this chart of a db for an app still under development (no usage expected other than when I'm working on it).
<img width="608" alt="image" src="https://github.com/dotnet/aspire-samples/assets/6483057/2f9d2c9d-642c-4543-b6d5-3923d6b37bf0">

Does Aspire have tooling to provision a 'Job' type yet?

If not, maybe we update this sample to prevent surprises? Something like the below [here](https://github.com/dotnet/aspire-samples/blob/ed1582c9fa890363ab9590bb553f12fa969eba7b/samples/DatabaseMigrations/DatabaseMigrations.AppHost/Program.cs#L11)?

```csharp
if (!builder.ExecutionContext.IsPublishMode) {
    builder.AddProject<Projects.DatabaseMigrations_MigrationService>("migration")
           .WithReference(db1);
}
```

**Comment by DamianEdwards** [comment]
I think the point of the sample is to have the migration app deployed though, so that it can be responsible for running migrations in the published environment too. Adding @JamesNK and @AndriySvyryd who worked on that sample.

**Comment by JamesNK** [comment]
The intent of the app is it starts, runs any data migration, then shuts down to release resources. This is how it works in local development. That seems like the best behavior to have in ACA (or k8s)

ACA restarts the container when the app exits? Can a job be a container?

Perhaps Aspire need metadata to say that an app is job or shouldn't attempt to restart. That would then flow through to deployment.

**Comment by collinbarrett** [comment]
Right, on deploy to Azure, having it run once and exit until the next artifact change sounds good to me.

The issue is by default it was deploying as an "App" type which auto-restarts if it exits. If we can somehow flip it to "Job" type, I think that might be a better fit. See [these docs](https://learn.microsoft.com/en-us/azure/container-apps/jobs?tabs=azure-cli#compare-container-apps-and-jobs).

**Comment by DamianEdwards** [comment]
FYI @davidfowl @ellismg @mitchdenny as this relates to tweaking ACA properties for project resources.

**Comment by mitchdenny** [comment]
For the samples specifically I think there is a tension between optimizing for education (pulling some code into its own project) and optimizing for cost efficiency (running this code as a background service in a component you are already deploying anyway).

Supporting container jobs in ACA is definitely something we want to do.

**Comment by christiannagel** [comment]
I had the same issue with the restart of the initialization services and disabled this service after it was running.

With Azure Container Apps, *init containers* would be a great match for initialization scenarios such as creating the database:
[Add support for Azure Container Apps "init containers"](https://github.com/dotnet/aspire/issues/2866)

**Comment by parched** [comment]
Why wouldn't the migration service be run at the API service startup? That would simplify this sample and fix this issue.

**Comment by DamianEdwards** [comment]
We now support ACA jobs so we could update this sample's readme to demonstrate how to configure the migration to run as a job.
